src/utils/index.tsx

```diff
@@ -0,0 +1,264 @@
+//     )
+//       .then(response => response.json())
+//       .then(data => {
+//         return data?.result
+//       })
+
+//     return await fetch(
+//       `https://${api}.etherscan.io/api?module=account&action=txlist&address=${address}&startblock=0&endblock=99999999&sort=asc&apikey=${apiKey}`
+//     )
+//       .then(response => response.json())
+//       .then(async data => {
+//         const modified = []
+//         if (data?.status === '0' || typeof data?.result === 'string') {
+//           return {
+//             balance,
+//             txs: [],
+//             error: data?.result
+//           }
+//         }
+//         for (const t of data?.result) {
+//           const extra = apolloClient
+//             ? await apolloClient?.query({
+//                 query: isDonor ? GET_USER_BY_ADDRESS : GET_PROJECT_BY_ADDRESS,
+//                 constiables: {
+//                   address: isDonor
+//                     ? Web3.utils.toChecksumAddress(t?.from)
+//                     : Web3.utils.toChecksumAddress(t?.to)
+//                 }
+//               })
+//             : null
+//           modified.push({
+//             ...t,
+//             extra: extra?.data || null,
+//             donor: t.from,
+//             createdAt: t.timeStamp,
+//             currency: 'ETH'
+//           })
+//         }
+//         return {
+//           balance,
+//           txs: modified
+//         }
+//       })
+//   } catch (error) {
+//     console.log({ error })
+//   }
+// }
+
+export const getERC20List = ERC20List;
+
+export async function checkIfURLisValid(checkUrl: string) {
+	let url = checkUrl;
+	if (!/^(?:f|ht)tps?:\/\//.test(checkUrl)) {
+		url = 'https://' + url;
+	}
+	const pattern = new RegExp(
+		'^(https?:\\/\\/)?' + // protocol
+			'((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+			'((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+			'(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+			'(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+			'(\\#[-a-z\\d_]*)?$',
+		'i',
+	);
+	return !!pattern.test(url);
+}
+
+export const fetchPrices = (
+	chain: any,
+	tokenAddress: any,
+	catchFunction: any,
+) => {
+	return fetch(
+		`https://api.coingecko.com/api/v3/simple/token_price/${chain}?contract_addresses=${tokenAddress}&vs_currencies=usd`,
+	)
+		.then(response => response.json())
+		.then(data => parseFloat(data[Object.keys(data)[0]]?.usd?.toFixed(2)))
+		.catch(err => {
+			console.log('Error fetching prices: ', err);
+			catchFunction(0);
+		});
+};
+
+export const fetchEthPrice = (catchFunction: any) => {
+	return fetch(
+		'https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd',
+	)
+		.then(response => response.json())
+		.then(data => data.ethereum.usd)
+		.catch(err => {
+			console.log('Error fetching ETH price: ', err);
+			catchFunction(0);
+		});
+};
```

Comment:
I think we can change usePrice and use it instead of this. ( we get eth price it there).

Reply:
But I think that price relates only to the main chain token (ETH/xDAI) how can I use it for any token?